devtools/client/framework/target-from-url.js

```diff
@@ -71,7 +74,7 @@ exports.targetFromURL = Task.async(function*(url) {
     // Fetch target for a remote chrome actor
     DebuggerServer.allowChromeProcess = true;
     try {
-      id = parseInt(id);
+      id = parseInt(id, 10);
```

Comment:
There is a lot of redundancy wrt the id in this code and below. Why not deal with it on line 46, where id is first set up? At least there, it could be parseInt()'d to get either the NaN or number. Then we wouldn't need to do this check after each `if`. It might help flatten the code a bit.

Also, because "type" is working as a enum, this code is kinda screaming to be made into a switch block.